IOPool/Output/src/RootOutputTree.cc

```diff
@@ -181,8 +178,8 @@ namespace edm {
     if(inputTree == nullptr) return false;
 
     // Do the sub-branches match in the input and output. Extra sub-branches in the input are OK for fast cloning, but not in the output.
-    for(std::vector<TBranch*>::const_iterator it = readBranches_.begin(), itEnd = readBranches_.end(); it != itEnd; ++it) {
-      TBranchElement* outputBranch = dynamic_cast<TBranchElement*>(*it);
+    for(auto const& outputBr : readBranches_) {
+      TBranchElement* outputBranch = dynamic_cast<TBranchElement*>(outputBr);
```

Comment:
@Dr15Jones Not really. In the C++ meaning of constness, which does not propagate, this cast cannot cast away constness. In the propagating meaning of constness, this was not const in the first place because propagate_const is not implemented yet for readBranches_. It was one of the "hard cases."